packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -130,14 +192,317 @@ function isEmptyConfig({
   );
 }
 
-export class IrisGrid extends Component {
+type Settings = {
+  defaultDateTimeFormat?: string;
+  defaultDecimalFormatOptions?: {
+    defaultFormatString?: string;
+  };
+  defaultIntegerFormatOptions?: {
+    defaultFormatString?: string;
+  };
+  formatter?: FormattingRule[];
+  timeZone?: string;
+  showTimeZone?: boolean;
+  showTSeparator?: boolean;
+  truncateNumbersWithPound?: boolean;
+  disableMoveConfirmation?: boolean;
+  showSystemBadge?: boolean;
+};
```

Comment:
We should only need to define the settings that IrisGrid actually uses here. So I think we can drop `disableMoveConfirmation` and `showSystemBadge` for example.
```suggestion
type Settings = {
  defaultDateTimeFormat?: string;
  defaultDecimalFormatOptions?: {
    defaultFormatString?: string;
  };
  defaultIntegerFormatOptions?: {
    defaultFormatString?: string;
  };
  formatter?: FormattingRule[];
  timeZone?: string;
  showTimeZone?: boolean;
  showTSeparator?: boolean;
  truncateNumbersWithPound?: boolean;
};
```